### DISCUSSION TITLE
customize background in layout

### DISCUSSION THREAD

**Discussion by nico1080** [mattblovell/kodi_panel#18] [Ideas]
By default, layouts have back background and are populated with various item (info label, lables, thumb, etc)

Having the ability, to change background could be nice.
Exemple:

- change backgound color, 
- draw simple geometry (frames)
- add background image

**Comment by mattblovell** [comment]
Both `video_screen_static()` and `audio_screen_static()` start by invoking Pillow's `Image.new()`.  The color that gets passed to that call could be pulled from the layout's dictionary.

Additional options could also permit for invoking `draw.rectangle()`, as occurs at the beginning of `update_display()` or for pasting an image.  Presumably we could rely upon the specified background image already being appropriately sized for the display.  The rectangle / frame option would need options for frame color, interior color, and frame width, corresponding to the `outline`, `fill`, and `width` arguments to ImageDraw.rectangle:

https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.rectangle

Based on the artwork rendering speed for cover art using an RPi Zero on an 800x480 display, filling in an entire background image may not be particularly speedy.  More capable Pi models would likely not have any issue, though.  If a single background image was going to be used for all screens, that could save some performance, but I think the idea is that every layout could use something different.

**Comment by mattblovell** [comment]
Implemented in v1.26
